New pickling methods loads all absolute links

Since the PR https://github.com/stac-utils/pystac/pull/1285 and more specifically this [change](https://github.com/stac-utils/pystac/pull/1285/files#diff-c235f14269d2641c490a6ec16abb7ab96c1f7d4e2ca4514b0ade666b2094469eR179), it seems that a deepcopy of an item will load all the links with an absolute href.
Is it intended or am I missing something?
This causes issue when loading assets using `get_assets` method that makes first a deep copy of the stac object that uses pickling.
When I load an item with unreachable links (e.g. s3 url but no custom IO reader set) and try to list the assets, it raises an issue.

```rio_tiler/io/stac.py:258: in __attrs_post_init__
    self.assets = list(
rio_tiler/io/stac.py:149: in _get_assets
    for asset, asset_info in stac_item.get_assets().items():
venv/lib/python3.11/site-packages/pystac/asset.py:300: in get_assets
    return {
venv/lib/python3.11/site-packages/pystac/asset.py:301: in <dictcomp>
    k: deepcopy(v)
/usr/lib/python3.11/copy.py:172: in deepcopy
    y = _reconstruct(x, memo, *rv)
/usr/lib/python3.11/copy.py:271: in _reconstruct
    state = deepcopy(state, memo)
/usr/lib/python3.11/copy.py:146: in deepcopy
    y = copier(x, memo)
/usr/lib/python3.11/copy.py:231: in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
/usr/lib/python3.11/copy.py:161: in deepcopy
    rv = reductor(4)
venv/lib/python3.11/site-packages/pystac/item.py:179: in __getstate__
    d["links"] = [
venv/lib/python3.11/site-packages/pystac/item.py:180: in <listcomp>
    link.to_dict() if link.get_href() else link for link in d["links"]
venv/lib/python3.11/site-packages/pystac/link.py:181: in get_href
    and self.owner.get_root()
venv/lib/python3.11/site-packages/pystac/stac_object.py:326: in get_root
    root_link.resolve_stac_object()
venv/lib/python3.11/site-packages/pystac/link.py:330: in resolve_stac_object
    obj = stac_io.read_stac_object(target_href, root=root)
venv/lib/python3.11/site-packages/pystac/stac_io.py:234: in read_stac_object
    d = self.read_json(source, *args, **kwargs)
venv/lib/python3.11/site-packages/pystac/stac_io.py:205: in read_json
    txt = self.read_text(source, *args, **kwargs)
venv/lib/python3.11/site-packages/pystac/stac_io.py:282: in read_text
    return self.read_text_from_href(href)
venv/lib/python3.11/site-packages/pystac/stac_io.py:300: in read_text_from_href
    with urlopen(req) as f:
/usr/lib/python3.11/urllib/request.py:216: in urlopen
    return opener.open(url, data, timeout)
/usr/lib/python3.11/urllib/request.py:519: in open
    response = self._open(req, data)
/usr/lib/python3.11/urllib/request.py:541: in _open
    return self._call_chain(self.handle_open, 'unknown',
/usr/lib/python3.11/urllib/request.py:496: in _call_chain
    result = func(*args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <urllib.request.UnknownHandler object at 0x74a76915ba50>
req = <urllib.request.Request object at 0x74a768cd8dd0>

    def unknown_open(self, req):
        type = req.type
>       raise URLError('unknown url type: %s' % type)
E       urllib.error.URLError: <urlopen error unknown url type: s3>
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+### Fixed
+
+- Don't transform hrefs in `Item.__getstate__` ([#1337](https://github.com/stac-utils/pystac/pull/1337))
+
 ## [v1.10.0] - 2024-03-28
 
 ### Added

--- a/pystac/item.py
+++ b/pystac/item.py
@@ -177,7 +177,12 @@ class Item(STACObject, Assets):
         d = self.__dict__.copy()
 
         d["links"] = [
-            link.to_dict() if link.get_href() else link for link in d["links"]
+            (
+                link.to_dict(transform_href=False)
+                if link.get_href(transform_href=False)
+                else link
+            )
+            for link in d["links"]
         ]
 
         return d

--- a/tests/test_item.py
+++ b/tests/test_item.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import copy
 import json
 import os
 import pickle
@@ -682,3 +683,12 @@ def test_pickle_with_only_href_links(item: Item) -> None:
         assert original.media_type == new.media_type
         assert str(original.owner) == str(new.owner)
         assert str(original.target) == str(new.target)
+
+
+def test_copy_with_unresolveable_root(item: Item) -> None:
+    item.add_link(
+        pystac.Link(
+            "root", "s3://naip-visualization/this-is-a-non-existent-catalog.json"
+        )
+    )
+    copy.deepcopy(item)

